src/modules/musig/tests_impl.h

```diff
@@ -921,6 +923,367 @@ void musig_tweak_test(secp256k1_scratch_space *scratch) {
     }
 }
 
+int musig_vectors_keyagg_and_tweak(secp256k1_musig_keyagg_cache *keyagg_cache,
+                                   unsigned char *agg_pk_ser,
+                                   unsigned char pubkeys33[][33],
+                                   unsigned char tweaks32[][32],
+                                   size_t key_indices_len,
+                                   size_t *key_indices,
+                                   size_t tweak_indices_len,
+                                   size_t *tweak_indices,
+                                   int *is_xonly) {
+    secp256k1_pubkey pubkeys[MUSIG_VECTORS_MAX_PUBKEYS];
+    const secp256k1_pubkey *pk_ptr[MUSIG_VECTORS_MAX_PUBKEYS];
+    int i;
+    secp256k1_pubkey agg_pk;
+    secp256k1_xonly_pubkey agg_pk_xonly;
+
+    for (i = 0; i < (int)key_indices_len; i++) {
+        if (!secp256k1_ec_pubkey_parse(ctx, &pubkeys[i], pubkeys33[key_indices[i]], 33)) {
+            return 0;
+        }
+        pk_ptr[i] = &pubkeys[i];
+
+    }
+    if (!secp256k1_musig_pubkey_agg(ctx, NULL, NULL, keyagg_cache, pk_ptr, key_indices_len)) {
+        return 0;
```

Comment:
is there a reason why you return 0 in this function (instead of CHECK)? CHECK makes it easier to spot the failing line in case of failure

Reply:
Yes, there is a reason. For some test vectors I want the function to fail (`CHECK(!musig_vectors_keyagg_and_tweak(...)`) and therefore do not want to abort the program.

Reply:
I changed the function such that also error enum is set to distinguish the various error cases in keyagg_and_tweak.